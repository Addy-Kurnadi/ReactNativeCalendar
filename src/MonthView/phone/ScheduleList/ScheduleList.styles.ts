import { StyleSheet, TextStyle, ViewStyle } from "react-native"

interface Style {
    scrollView: ViewStyle
    textStyle: TextStyle
    leftEventHeader: ViewStyle
    rightEventHeader: ViewStyle
    outerViewContainer: ViewStyle
}

export default StyleSheet.create<Style>({
    scrollView: {
        flexDirection: "column"
    },
    textStyle: {
        paddingLeft: 10,
        fontSize: 14,
        fontWeight: "bold",
        color: "black"
    },
    leftEventHeader: {
        flex: 1
    },
    rightEventHeader: {
        flex: 4,
    },
    outerViewContainer: {
        flex: 1,
        flexDirection: "row",
        justifyContent: "center"
    }
})
