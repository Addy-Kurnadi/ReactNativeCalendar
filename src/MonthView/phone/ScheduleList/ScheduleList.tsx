import * as React from "react"
import { ScrollView, Text, View } from "react-native"
import styles from "./ScheduleList.styles"

import { ScheduleListOutline } from "./ScheduleListOutline"
import { Timeline } from "./Timeline"

interface Props {
    startTime: any
    endTime: any
    height: number
}

export const ScheduleList = ({startTime, endTime, height}: Props) => {
    const numberToRender = (endTime - startTime) / 100
    return (
        <ScrollView>
            <View style={{flexDirection: "row"}}>
                <View style={styles.leftEventHeader}/>
                <View style={styles.rightEventHeader}>
                    <Text style={styles.textStyle}>{"Event"}</Text>
                </View>
            </View>
            <View style={styles.outerViewContainer}>
                <Timeline startTime={startTime} endTime={endTime} height={height}/>
                <ScheduleListOutline height={height} numberToRender={numberToRender}/>
            </View>
        </ScrollView>
    )
}
