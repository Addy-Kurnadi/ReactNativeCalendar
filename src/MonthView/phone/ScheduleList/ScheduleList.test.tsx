import * as React from "react"
import { create } from "react-test-renderer"
import { ScheduleList } from "./ScheduleList"

describe("<Timeline />", () => {
    it("should render with given props", () => {
        const component = create(
            <ScheduleList
                startTime={"0800"}
                endTime={"1800"}
                height={50}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
})
