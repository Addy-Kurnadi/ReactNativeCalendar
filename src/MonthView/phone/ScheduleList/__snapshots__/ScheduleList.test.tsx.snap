// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render with given props 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 4,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 14,
              "fontWeight": "bold",
              "paddingLeft": 10,
            }
          }
        >
          Event
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          08:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          09:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          10:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          11:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          12:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          13:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          14:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          15:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          16:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          17:00
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "marginVertical": 16.5,
              },
            ]
          }
        >
          18:00
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 25,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "borderBottomColor": "#d3d3d3",
                "borderBottomWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 50,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderLeftColor": "#d3d3d3",
                "borderLeftWidth": 1,
                "borderStyle": "solid",
              },
              Object {
                "height": 25,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderColor": "#d3d3d3",
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "position": "absolute",
                },
                Object {
                  "borderRadius": 5,
                  "height": 10,
                  "left": -5,
                  "top": -5,
                  "width": 10,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
